easybuild/easyconfigs/h/HOOMD-blue/HOOMD-blue-4.9.1-foss-2023a.eb

```diff
@@ -0,0 +1,73 @@
+# LLVM included, needs to be version 16.x as newer versions have a different API
+# Author: J. Sassmannshausenb (Imperial College London/UK)
+
+easyblock = 'CMakeMake'
+
+name = 'HOOMD-blue'
+version = "4.9.1"
+
+homepage = "https://bitbucket.org/glotzer/hoomd-blue"
+description = """HOOMD-blue is a general-purpose particle simulation
+toolkit, implementing molecular dynamics and hard particle Monte Carlo
+optimized for fast execution on both GPUs and CPUs."""
+
+toolchain = {'name': 'foss', 'version': '2023a'}
+toolchainopts = {'usempi': True}
+
+github_account = 'glotzerlab'
+source_urls = [GITHUB_LOWER_RELEASE]
+sources = ['hoomd-%(version)s.tar.gz']
+patches = ['%(name)s-%(version)s.patch']
+checksums = [
+    {'hoomd-4.9.1.tar.gz': 'b48ff3d44fef56ddf95c64b462b5d7f670bfc7a76b47223207edd3e7438e52bd'},
+    {'HOOMD-blue-4.9.1.patch': '3b611c177414c2b9c1da1c7dd54d149406875fda37f92fcf24ce34ebf416a30b'},
+]
+
+builddependencies = [
+    ('CMake', '3.26.3'),
+    ('pybind11', '2.11.1'),
+    ('Ninja', '1.11.1'),
+]
+
+dependencies = [
+    ('Python', '3.11.3'),
+    ('SciPy-bundle', '2023.07'),
+    ('tbb', '2021.11.0'),
+    ('Eigen', '3.4.0'),
+    ('Cereal', '1.3.2', '', SYSTEM),
+    ('LLVM', '16.0.6'),
+    ('Clang', '16.0.6'),
```

Comment:
Clang dependency will no longer be required with an LLVM built with https://github.com/easybuilders/easybuild-easyblocks/pull/3373

Reply:
That might be true for newer versions but I am stuck with the `16.0.6` version **and** I need `Clang`. So in the end, the way forward was to use the `Clang` included `LLVM` but build `Clang` with shared libraries. Then it was very easy.

Reply:
I've tried quite a lot of things and it simply isnä't possible to build HOOMD-blue with the existing statical libarary Clang 16.0.6
Once the _jit...so python library is linked with all libraries it needs it fails with
```
== FAILED: Installation ended unsuccessfully: Sanity check failed: sanity check command python -c 'import hoomd' failed with exit code -6 (output: : CommandLine Error: Option 'scalarize-variable-insert-extract' registered more than once!
```
That comes from LLVM somewhere, and it just isn't possible to get rid of with that build. It do need a shared library Clang build.